ps_checkout.php

```diff
@@ -212,11 +213,50 @@ public function uninstallTabs()
         return $uninstallTabCompleted;
     }
 
+    /**
+     * Express checkout on the first step of the checkout
+     * Used before 1.7.6 - hook DisplayPersonalInformationTop not available
+     */
+    public function hookHeader()
+    {
+        if (version_compare(_PS_VERSION_, '1.7.6.0', '>=')) {
+            return false;
+        }
+
+        $currentPage = $this->context->controller->php_self;
+
+        if ($currentPage != 'order') {
+            return false;
+        }
+
+        if (!(bool) Configuration::get('PS_CHECKOUT_EC_CHECKOUT_PAGE')) {
+            return false;
+        }
+
+        // Check if we are already in an express checkout
+        if (isset($this->context->cookie->paypalOrderId)) {
+            return false;
+        }
+
+        if (true === $this->isPaymentStep()) {
+            return false;
+        }
+
+        $expressCheckout = new ExpressCheckout($this, $this->context);
+        $expressCheckout->setDisplayMode(ExpressCheckout::CHECKOUT_MODE);
+
+        return $expressCheckout->render();
```

Comment:
Instead of copy pasting code, please create a private method to call in your two hooks.